pages/[username]/index.public.js

```diff
@@ -173,31 +196,233 @@ export default function Page({ userFound: userFoundFallback }) {
         </Box>
       </Box>
 
-      {userFound.description && (
-        <Box
-          sx={{
-            borderWidth: 1,
-            borderStyle: 'solid',
-            borderColor: 'border.default',
-            borderRadius: '6px',
-            width: '100%',
-            p: 3,
-            mb: 3,
-            overflow: 'hidden',
-          }}>
-          <Viewer value={userFound.description} />
-        </Box>
+      {userFound.description && !isEditingDescription && (
+        <>
+          <Box
+            sx={{
+              display: 'flex',
+              width: '100%',
+              justifyContent: 'space-between',
+              mb: 1,
+              minHeight: '32px', // This is the button height, so it doesn't flick when the button is rendered
+            }}>
+            <Text sx={{ fontSize: 1, fontWeight: 'bold', mt: 'auto', pb: 1 }}>Descrição</Text>
+            {canUpdateDescription && (
+              <Button variant="invisible" onClick={handleEditDescription}>
+                Editar descrição
+              </Button>
+            )}
+          </Box>
+
+          <Box
+            sx={{
+              borderWidth: 1,
+              borderStyle: 'solid',
+              borderColor: 'border.default',
+              borderRadius: 2,
+              width: '100%',
+              p: 3,
+              overflow: 'hidden',
+            }}>
+            <Viewer value={userFound.description} />
+          </Box>
+        </>
       )}
 
-      {!userFound.description && canUpdate && (
-        <Button href="/perfil" as={NextLink} sx={{ mx: 'auto' }}>
+      {!userFound.description && canUpdateDescription && !isEditingDescription && (
+        <Button sx={{ mx: 'auto', mt: 1 }} onClick={handleEditDescription}>
           Criar descrição
         </Button>
       )}
+
+      {isEditingDescription && (
+        <DescriptionForm
+          user={userFound}
+          onCancel={() => setIsEditingDescription(false)}
+          onSuccess={handleEditSuccess}
+          globalMessageObject={globalMessageObject}
+          setGlobalMessageObject={setGlobalMessageObject}
+          isAuthenticatedUser={isAuthenticatedUser}
+        />
+      )}
+
+      {!isEditingDescription && globalMessageObject?.position === 'description' && (
+        <Flash variant={globalMessageObject.type} sx={{ width: '100%', mt: 3 }}>
+          {globalMessageObject.text}
+        </Flash>
+      )}
     </DefaultLayout>
   );
 }
 
+function DescriptionForm({
+  user,
+  onCancel,
+  onSuccess,
+  globalMessageObject,
+  setGlobalMessageObject,
+  isAuthenticatedUser,
+}) {
+  const [description, setDescription] = useState(user.description ?? '');
+  const [isPosting, setIsPosting] = useState(false);
+  const [errorObject, setErrorObject] = useState();
+
+  const confirm = useConfirm();
+
+  async function handleSubmit(event) {
```

Comment:
A outra sugestão elimina esta necessidade, mas eu manteria apenas por precaução.

```suggestion
  const confirm = useConfirm();
  
  useEffect(() => {
    setDescription(user.description);
  }, [user]);

  async function handleSubmit(event) {
```

Reply:
Ainda não testei, mas acho que ao invés disso, podemos passar ```key={`description-form-${userFound.id}`}``` para o `DescriptionForm`. O que acha? Assim evita outros possíveis problemas.

Quando coloquei o `useEffect`, pensava que a melhor solução era passar uma `key` para o componente, só que o componente era a própria página (`Page`) então imaginei não ser possível, porque não sei como funciona exatamente essa parte da interação Next-React. Olhando melhor agora, poderia experimentar passar na linha `490`. Se funcionar, não precisaria de `key` no `DescriptionForm`.

Reply:
Fiz a [alteração](https://github.com/filipedeschamps/tabnews.com.br/compare/8d3dcade538d98ab2bc0bfe336d5bb23cddbe64b..29e4bae75849ba18e714010838b06c1b8e5d3f2f) que comentei. Se achar que ficou bom, pode realizar o merge.